datastore/store.go

```diff
@@ -43,14 +43,44 @@ type CachedDB struct {
 	malfeasanceCache *lru.Cache[types.NodeID, *types.MalfeasanceProof]
 }
 
+type Config struct {
+	ATXSize         int `mapstructure:"atx-size"`
+	MalfeasenceSize int `mapstructure:"malfeasence-size"`
+}
+
+func DefaultConfig() Config {
+	return Config{
+		ATXSize:         10_000,
+		MalfeasenceSize: 1_000,
+	}
+}
+
+type cacheOpts struct {
+	cfg Config
+}
+
+type Opt func(*cacheOpts)
+
+func WithConfig(cfg Config) Opt {
+	return func(o *cacheOpts) {
+		o.cfg = cfg
+	}
+}
+
 // NewCachedDB create an instance of a CachedDB.
-func NewCachedDB(db *sql.Database, lg log.Log) *CachedDB {
-	atxHdrCache, err := lru.New[types.ATXID, *types.ActivationTxHeader](atxHdrCacheSize)
```

Comment:
It seems that `atxHdrCacheSize` is still used but only for the `vrfNonceCache`, so it could be renamed.